### PR TITLE
layers: Warn if no present modes provided to create swapchain

### PR DESCRIPTION
**Pull request by HildarTheDorf**
This is needed by VK_EXT_swapchain_maintenance1 if using per-present-mode capabilities as documented [here](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkSurfacePresentModeEXT.html).

### CODE REVIEW

File: layers/core_checks/cc_wsi.cpp [hunk 1]
```diff
@@ -722,6 +722,17 @@ bool CoreChecks::ValidateCreateSwapchain(VkSwapchainCreateInfoKHR const *pCreate
             return true;
         }
     }
+
+    if (IsExtEnabled(device_extensions.ext_swapchain_maintenance1) && !LvlFindInChain<safe_VkSwapchainPresentModesCreateInfoEXT>(pCreateInfo->pNext)) {
+        if (LogWarning(device, "VUID-VkSwapchainCreateInfoKHR-pNext-00000",
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_wsi.cpp:727]: so this is a warning, this belongs in the Best Practice layer code `bp_wsi.cpp`

Comment (HildarTheDorf) [reply layers/core_checks/cc_wsi.cpp:727]: Ah, cool, I'll move it then.

File: tests/unit/best_practices.cpp [hunk 2]
```diff
@@ -2378,4 +2378,23 @@ TEST_F(VkBestPracticesLayerTest, NonSimultaneousSecondaryMarksPrimary) {
...
+    ASSERT_NO_FATAL_FAILURE(InitBestPracticesFramework());
+    if (!AreRequiredExtensionsEnabled()) {
+        GTEST_SKIP() << RequiredExtensionsNotSupported() << " not supported.";
+    }
+
+    ASSERT_NO_FATAL_FAILURE(InitState());
+    if (!InitSurface()) {
+        GTEST_SKIP() << "Cannot create surface";
+    }
+    m_errorMonitor->SetDesiredFailureMsg(kWarningBit, kVUID_BestPractices_NoVkSwapchainPresentModesCreateInfoEXTProvided);
```
Comment (juan-lunarg) [inline tests/unit/best_practices.cpp:2397]: Sorry took me so long to respond.

`scripts/vk_validation_stats.py` is crashing here (Unhelpfully). The fix is to use the underlying string.

```
 python scripts/vk_validation_stats.py external/Debug/64/Vulkan-Headers/build/install/share/vulkan/registry/validusage.json -export_header
Unable to categorize VUID: 
Confused while parsing VUIDs in test code - cannot proceed. (FIXME)
```

```suggestion
    m_errorMonitor->SetDesiredFailureMsg(
        kWarningBit, "UNASSIGNED-BestPractices-vkCreateSwapchainKHR-no-VkSwapchainPresentModesCreateInfoEXT-provided");
```

**Review by arno-lunarg — APPROVED** [review]

**Comment by arno-lunarg** [comment]
Thank you for your contribution @HildarTheDorf !

**Pull request merged**